### PR TITLE
Issue #50: boolean functions - 1

### PR DESCRIPTION
**Pull request by increasechief**
## Summary

Adds test functions with boolean return statements rather then calling domain_error like existing check_* functions.

https://github.com/stan-dev/math/issues/50

## Tests

Each function included has its own test file modeled off the existing tests used for the check functions.

## Side Effects

None applicable.

## Checklist

- [x] Math issue #(50)

- [x] Copyright holder: (Same as source)

    The copyright holder is typically you or your assignee, such as a university or company. By submitting this pull request, the copyright holder is agreeing to the license the submitted work under the following licenses:
      - Code: BSD 3-clause (https://opensource.org/licenses/BSD-3-Clause)
      - Documentation: CC-BY 4.0 (https://creativecommons.org/licenses/by/4.0/)

- [x] the basic tests are passing

    - unit tests pass (to run, use: `./runTests.py test/unit`)
    - header checks pass, (`make test-headers`)
    - docs build, (`make doxygen`)
    - code passes the built in [C++ standards](https://github.com/stan-dev/stan/wiki/Code-Quality) checks (`make cpplint`)

- [x] the code is written in idiomatic C++ and changes are documented in the doxygen

- [x] the new changes are tested

### CODE REVIEW

**Comment by syclik** [comment]
@increasechief, on the previous PR, I left some comments regarding doc and I realized after the fact that the motivation probably wasn't clear. I wanted to run doxygen on our code base to show you the difference, but we have errors on `develop` at the moment. (It was the use of `@return` for the first line as opposed to just a sentence in plain English.) But I just wanted to say that's the reason for it... it's a lot easier to read the doc that way.

When the develop branch is working again, I'll put up a screenshot so we can take a look. I'm assuming I know how it's rendered, but maybe the way you had it renders fine.

File: stan/math/prim/arr.hpp [hunk 1]
```diff
@@ -14,6 +14,11 @@
 #include <stan/math/prim/arr/err/check_matching_sizes.hpp>
 #include <stan/math/prim/arr/err/check_nonzero_size.hpp>
 #include <stan/math/prim/arr/err/check_ordered.hpp>
+
+#include <stan/math/prim/arr/err/is_matching_size.hpp>
```
Comment (syclik) [inline stan/math/prim/arr.hpp]: can we alphabetize and drop the surrounding whitespace?

More concretely, 
```
#ifdef STAN_OPENCL
#include <stan/math/prim/arr/err/check_opencl.hpp>
#endif
```
should go before the `is_*` functions.

File: stan/math/prim/scal.hpp [hunk 2]
```diff
@@ -59,6 +59,8 @@
 #include <stan/math/prim/scal/err/invalid_argument_vec.hpp>
 #include <stan/math/prim/scal/err/out_of_range.hpp>
 
+#include <stan/math/prim/scal/err/is_size_match.hpp>
```
Comment (syclik) [inline stan/math/prim/scal.hpp]: alphabetize this include. It should go after `invalid_argument_vec.hpp`. And remove the surrounding whitespace.

**Comment by increasechief** [comment]
> @increasechief, on the previous PR, I left some comments regarding doc and I realized after the fact that the motivation probably wasn't clear. I wanted to run doxygen on our code base to show you the difference, but we have errors on `develop` at the moment. (It was the use of `@return` for the first line as opposed to just a sentence in plain English.) But I just wanted to say that's the reason for it... it's a lot easier to read the doc that way.
> 
> When the develop branch is working again, I'll put up a screenshot so we can take a look. I'm assuming I know how it's rendered, but maybe the way you had it renders fine.

**Comment by increasechief** [comment]
The doxygen formatting I was using was the same as the existing `check_` functions. I don't mind looking into possible changes for that with you. 

I've fixed those minor changes so that everything for the rest of the files in `mat` and `scal` should be consistent to what is here. I'm working on the `*_test.cpp` files at the moment which is relatively slow.

I'll close this branch (again) and start one with only a single commit to keep things clean. I hope to have the other work upload with a single attempt in the future.

**Comment by syclik** [comment]
Don’t worry too much about a single commit; it’s only really used when we
look at individual commits for comparison.
